templates/app/flake.nix

```diff
@@ -57,10 +57,14 @@
 
     in
     {
+      # 'app' is the name in pyproject.toml after name normalization.
+      # See https://packaging.python.org/en/latest/specifications/name-normalization/#normalization
+
       packages.x86_64-linux.default = python.pkgs.app;
-      # TODO: A better mkShell withPackages example.
+      # a development shell where python has the app package
+      # installed
       devShells.x86_64-linux.default = pkgs.mkShell {
-        inputsFrom = [ python.pkgs.app ];
+        buildInputs = [ (python.withPackages (p: [ p.app ])) ];
```

Comment:
These are not quite the same:

- `inputsFrom = [ python.pkgs.app ];`

Gives you a development shell _for_ `app`.

- buildInputs = [ (python.withPackages (p: [ p.app ])) ];

Gives you a development shell _with_ `app`

Reply:
What I _actually_ want here is an example for an editable package, but nixpkgs doesn't have support for that yet.
I'm planning to tackle that soon, but it hasn't been top priority yet.

Reply:
Fair. I'll reduce the PR to the comment about 'app' then (because I spent two whole minutes figuring out what 'app' was).